portfolio/src/main/webapp/script.js

```diff
@@ -287,10 +328,30 @@ async function addMarker(position) {
     const queryURL = "/markers?latitude=" + lat.toString() + "&longitude=" + lng.toString();
     const request = new Request(queryURL, {method: "POST"});
     const response = await fetch(request);
-    initMap();
+    displayMarkers();
 }
 
 async function initialize() {
     initMap();
     fetchCommentsWithStoredLimit();
+}
+
+async function deleteMarker(latitude, longitude) {
+    console.log("Delete the marker");
+    removeMarker(latitude, longitude);
+    const queryURL = "/markers?latitude=" + latitude.toString() + "&longitude=" + longitude.toString();
+    const request = new Request(queryURL, {method: "DELETE"});
+    const response = await fetch(request);
+}
+
+function removeMarker(latitude, longitude) {
+    for (var i = 0; i < markers.length; i++) {
+        var marker = markers[i];
+        if (marker.getPosition().lat() === latitude
+         && marker.getPosition().lng() === longitude) {
+             markers.splice(i, 1);
```

Comment:
I usually avoid modifying arrays that I'm iterating over inside the for-loop.

Alternatively, we can use [Array.filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to create a new list of markers, while also getting a chance to call methods on each element.